### PR TITLE
Graphs v2

### PR DESCRIPTION
**Pull request by madamczykhabana**
Rework of HPU graphs. Now the flow looks like this:
* VLLM_GRAPH_RESERVED_MEM is used to determine how much free memory after loading weights should be used for HPU-graphs (by default 30%)
* we allocate blocks according to gpu-memory-utilization
* we warmup all shapes without HPU graphs
* we calculate remaining free memory and split it between prompt and decode graphs according to VLLM_GRAPH_PROMPT_RATIO (50%) and VLLM_GRAPH_MEM_MARGIN (5%)
* we capture prompt graphs in order defined by bs*seq_len stoping when according to heuristics we won't fit another graph
* as above but for decode graphs

Other important changes:
* selecting token ids has been moved inside HPU graphs to limit memory usage
* calculating attention_mask has been moved inside HPU graphs, but before model.forward to limit memory usage
* *AttentionMetadata objects are trimmed before going into HPU graphs for better control over parameters (and avoiding recompilations due to changing constants)

### CODE REVIEW

**Comment by madamczykhabana** [comment]
@kzawora-intel Please review

File: vllm/worker/habana_model_runner.py [hunk 1]
```diff
@@ -77,6 +87,78 @@ def find_bucket(value: int, config: Tuple[int, int, int]):
...
+    def __init__(self, model):
+        self.model = model
+
+    def _set_attn_bias(self, attn_metadata, batch_size, seq_len, device, dtype):
+        prefill_metadata = attn_metadata.prefill_metadata
+        if prefill_metadata is None:
+            return attn_metadata
+        #FIXME: Restore alibi support
+        #if self.alibi_slopes is None:
+        if True:
```
Comment (kzawora-intel) [inline vllm/worker/habana_model_runner.py:120]: nit: can we throw an exception/make an assertion instead of if True?

Comment (madamczykhabana) [reply vllm/worker/habana_model_runner.py:120]: it's a bit of extra work as currently ModelAdapter has no access to alibi_slopes so we'd need to figure out first how to access that. At least in my head, fixing alibi is something that would probably need a better design anyway, so I'm not sure if it's worth fixing at this point.

**Review by kzawora-intel — APPROVED** [review]
looks very good, thanks!
